api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -339,6 +339,36 @@ message RequirementRule {
   JwtRequirement requires = 2;
 }
 
+// This message specifies Jwt requirements based on a Metadata string value
+// Other HTTP filters can use this message to specify Jwt requirement.
+//
+// Example:
+//
+// .. code-block:: yaml
+//
+//    filter: foo
+//    path:
+//      - jwt_selector
+//    requires:
+//      issuer_1:
+//        provider_name: issuer1
+//      issuer_2:
+//        provider_name: issuer2
+//
+// If the filter "foo" sets {"jwt_selector": "issuer_1"} to the dynamicMetadata for a request,
+// jwt_authn filter will use JwtRequirement{"provider_name": "issuer1"} to verify.
+message MetadataRule {
+  // The filter name to retrieve the dynamic metadata.
+  string filter = 1 [(validate.rules).string.min_bytes = 1];
+
+  // The path to retrieve a string value from the dynamic metadata.
+  repeated string path = 2 [(validate.rules).repeated .min_items = 1];
+
+  // A map of string keys to requirements. The string key is the value
+  // in the dynamic metadata specified by `filter` and `path` fields.
```

Comment:
Nit: `*filter*` and `*path*` (or link to the field with `:ref:`)

Reply:
Done